packages/qdrant-loader/tests/unit/core/worker/test_pool.py

```
@@ -260,3 +260,94 @@ async def handler(_job_type: str, _payload: dict) -> None:  # pragma: no cover
+    sqlite_job_queue: SQLiteJobQueue,
+):
+    """Renewal DB failures must not replace the handler exception.
+
+    If extend_visibility fails while handler is still running, final job failure
+    should reflect the handler error, not the renewal-task error.
+    """
+    await sqlite_job_queue.enqueue("INCREMENTAL_PULL", {"source_lock": "jira-main"})
+
+    original_extend_visibility = sqlite_job_queue.extend_visibility
+
+    async def flaky_extend_visibility(job_id: int, lease_seconds: int) -> bool:
```

Ruff (0.15.15)

[warning] Unused function argument: `job_id`

(ARG001)